lib/ChargeBee/Environment.php

```diff
@@ -30,6 +36,15 @@ public function __construct($site, $apiKey)
         } else {
             $this->apiEndPoint = self::$scheme . "://$site." . self::$chargebeeDomain . "/api/" . self::API_VERSION;
         }
+
+        self::$httpClientFactory = new GuzzleFactory(
```

Comment:
Since there is a setHttpClientFactory() setter, maybe the default "GuzzleFactory" value should not be instanciated in the __construct(), but rather in the getter if no facory was set ?

Reply:
It got me thinking, maybe I could use a named constructor with optional parameter and only then (if empty) set up default guzzle factory. Avoids possible state mutation.